spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorProperties.java

```diff
@@ -58,9 +58,9 @@
 	private String resultHeader;
 
 	/**
-	 * If not empty, the processor will use the inputHeader content as data input instead of message's payload.
+	 * Specifies where the input data should comes from. Defaults to {@link org.springframework.messaging.Message}'s payload.
 	 */
-	private String inputHeader;
+	private String inputExpression = "payload";
```

Comment:
Let's make it just as `expression`!
And you really can have it as a `Expression` type.
You don't need that `ExpressionEvaluatingMessageProcessor` at all in the `TensorflowProcessorConfiguration` above.

Also I think for better performance we should leave it as `null` by default and fallback to `message.getPayload()`.